diff -u -r libdrm-libdrm-2.4.100-old/configure.ac libdrm-libdrm-2.4.100/configure.ac
--- libdrm-libdrm-2.4.100-old/configure.ac	2019-10-17 00:35:55.000000000 +0300
+++ libdrm-libdrm-2.4.100/configure.ac	2026-07-12 13:06:07.816213829 +0300
@@ -44,13 +44,8 @@
 
 # Check for programs
 AC_PROG_CC
-AC_PROG_CC_C99
 AC_PROG_NM
 
-if test "x$ac_cv_prog_cc_c99" = xno; then
-	AC_MSG_ERROR([Building libdrm requires C99 enabled compiler])
-fi
-
 AC_USE_SYSTEM_EXTENSIONS
 AC_SYS_LARGEFILE
 AC_FUNC_ALLOCA
